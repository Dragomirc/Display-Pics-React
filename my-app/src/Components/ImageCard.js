import React from "react";
import styled from "styled-components";
const Card = styled.li`
  padding: 0.25rem;
  margin: 1rem;

  border: 3px dashed;

  cursor: pointer;
`;

const ImageCard = ({ previewURL, url, tags }) => {
  return (
    <Card>
      <a href={url}>
        <img src={previewURL} alt={`Reflecting ${tags}`} />
      </a>
    </Card>
  );
};

export default ImageCard;
