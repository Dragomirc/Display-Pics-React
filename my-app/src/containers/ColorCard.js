import React, { Component } from "react";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import styled, { keyframes } from "styled-components";
import { generateSearchTerm, searchTerms } from "../logic";
import { mapColor } from "../logic";
import { fetchPictures } from "../actions/index";

const fadein = keyframes`
  from { opacity:0; } to { opacity:1; }
`;
const Card = styled.li`
  padding: 1rem;
  margin: 1rem;
  border: 3px dashed white;
  width: 5em;
  height: 5em;
  cursor: pointer;
  animation: ${fadein} ease-in 1;
  animation-fill-mode: forwards;
  animation-duration: 1s;

  opacity: 0;
  &:nth-child(1) {
    animation-delay: 0.4s;
  }
  &:nth-child(2) {
    animation-delay: 0.9s;
  }
  &:nth-child(3) {
    animation-delay: 1.7s;
  }
  &:nth-child(4) {
    animation-delay: 2.4s;
  }
`;
class ColorCard extends Component {
  render() {
    return (
      <Card
        style={{ backgroundColor: mapColor(this.props.color) }}
        onClick={() =>
          this.props.fetchPictures(
            this.props.color,
            searchTerms[generateSearchTerm(searchTerms)]
          )
        }
      />
    );
  }
}

const mapDispatchToProps = dispatch => {
  return bindActionCreators({ fetchPictures }, dispatch);
};

export default connect(null, mapDispatchToProps)(ColorCard);
