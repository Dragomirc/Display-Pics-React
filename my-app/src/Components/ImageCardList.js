import React from "react";
import ImageCard from "./ImageCard";
import styled from "styled-components";

const List = styled.ul`
  display: flex;
  flex-wrap: wrap;
  list-style-type: none;
  justify-content: space-around;
`;
const ImageCardList = ({ images: { hits = [] } }) => {
  const images = hits
    .slice(0, 5)
    .map(({ previewURL, pageURL, id, tags }) => (
      <ImageCard previewURL={previewURL} key={id} url={pageURL} tags={tags} />
    ));

  return <List>{images}</List>;
};

export default ImageCardList;
