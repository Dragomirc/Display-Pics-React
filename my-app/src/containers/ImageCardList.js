import React, { Component } from "react";
import { connect } from "react-redux";
import ImageCard from "../components/ImageCard";
import styled from "styled-components";

const List = styled.ul`
  display: flex;
  flex-wrap: wrap;
  list-style-type: none;
  justify-content: space-around;
`;
class ImageCardList extends Component {
  renderImages = imagesArray =>
    imagesArray
      .slice(0, 5)
      .map(({ previewURL, pageURL, id, tags }) => (
        <ImageCard previewURL={previewURL} key={id} url={pageURL} tags={tags} />
      ));

  render() {
    console.log("pictures", this.props.pictures);
    return <List>{this.renderImages(this.props.pictures)}</List>;
  }
}
const mapStateToProps = ({ pictures }) => {
  return {
    pictures
  };
};
export default connect(mapStateToProps)(ImageCardList);
