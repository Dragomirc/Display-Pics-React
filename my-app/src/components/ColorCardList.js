import React from "react";
import ColorCard from "./ColorCard";
import styled from "styled-components";

const CardList = styled.ul`
  display: flex;
  flex-wrap: wrap;
  list-style-type: none;
  justify-content: space-around;
`;

const colors = ["red", "yellow", "green", "purple"];
const ColorCardList = ({ info }) => {
  const colorItems = colors.map(color => {
    return <ColorCard color={color} info={info} key={color} />;
  });

  return <CardList>{colorItems}</CardList>;
};

export default ColorCardList;
