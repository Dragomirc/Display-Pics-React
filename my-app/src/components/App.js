import React, { Component } from "react";

import ColorCardList from "./ColorCardList";
import ImageCardList from "./ImageCardList";
import Header from "./Header";

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      images: {}
    };
  }

  handleChange = images => {
    this.setState({ images });
  };

  render() {
    return (
      <div style={{ maxWidth: 640 }}>
        <Header />
        <div>
          <ColorCardList info={this.handleChange} />
        </div>
        <div>
          <ImageCardList images={this.state.images} />
        </div>
      </div>
    );
  }
}

export default App;
