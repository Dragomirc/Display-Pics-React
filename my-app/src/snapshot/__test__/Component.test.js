import React from "react";
import renderer from "react-test-renderer";
import ColorCardList from "../../Components/ColorCardList";
import ColorCard from "../../Components/ColorCard";
import ImageCard from "../../Components/ImageCard";
import ImageCardList from "../../Components/ImageCardList";
import App from "../../Components/App";

it("App renders without crashing", () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("App renders 4 x ColorCards", () => {
  const component = renderer
    .create(<ColorCardList color={"green"} info={jest.fn()} color={"green"} />)
    .toJSON();
  expect(component).toMatchSnapshot();
});

it("ColorCard renders an li element", () => {
  const component = renderer
    .create(<ColorCard style={"green"} info={jest.fn()} />)
    .toJSON();
  expect(component).toMatchSnapshot();
});

it("ImageCard renders an li with one image child and one anchor element", () => {
  const component = renderer.create(<ImageCard />).toJSON();
  expect(component).toMatchSnapshot();
});

it("App should render with 5 x ImageCard components when the state is updated", () => {
  const tree = renderer.create(<App />);
  tree.getInstance().handleChange(mockApiResponse);
  expect(tree.toJSON()).toMatchSnapshot();
});
const mockApiResponse = {
  totalHits: 5,
  hits: [
    {
      previewHeight: 107,
      likes: 9,
      favorites: 12,
      tags: "domestic, cat, little",
      webformatHeight: 458,
      views: 1205,
      webformatWidth: 640,
      previewWidth: 150,
      comments: 4,
      downloads: 743,
      pageURL:
        "https://pixabay.com/en/domestic-cat-little-cute-kitten-3133620/",
      previewURL:
        "https://cdn.pixabay.com/photo/2018/02/06/00/20/domestic-3133620_150.jpg",
      webformatURL:
        "https://pixabay.com/get/ea34b20c2ef6013ed95c4518b74b4f9fe175e5d604b014409df3c77da3ecb2_640.jpg",
      imageWidth: 4096,
      user_id: 752536,
      user: "Desertrose7",
      type: "photo",
      id: 3133620,
      userImageURL:
        "https://cdn.pixabay.com/user/2016/03/14/13-25-18-933_250x250.jpg",
      imageHeight: 2936
    },
    {
      previewHeight: 91,
      likes: 49,
      favorites: 28,
      tags: "cat, red, cute",
      webformatHeight: 392,
      views: 5097,
      webformatWidth: 640,
      previewWidth: 150,
      comments: 25,
      downloads: 3323,
      pageURL:
        "https://pixabay.com/en/cat-red-cute-mackerel-tiger-sweet-3080965/",
      previewURL:
        "https://cdn.pixabay.com/photo/2018/01/13/23/28/cat-3080965_150.jpg",
      webformatURL:
        "https://pixabay.com/get/ea35b90f21f2043ed95c4518b74b4f9fe175e5d604b014409df3c77da3ecb2_640.jpg",
      imageWidth: 4464,
      user_id: 686414,
      user: "Alexas_Fotos",
      type: "photo",
      id: 3080965,
      userImageURL:
        "https://cdn.pixabay.com/user/2018/02/07/15-07-53-496_250x250.jpg",
      imageHeight: 2740
    },
    {
      previewHeight: 120,
      likes: 206,
      favorites: 191,
      tags: "cat, red, cute",
      webformatHeight: 513,
      views: 35130,
      webformatWidth: 640,
      previewWidth: 150,
      comments: 24,
      downloads: 19979,
      pageURL:
        "https://pixabay.com/en/cat-red-cute-mackerel-tiger-sweet-1652880/",
      previewURL:
        "https://cdn.pixabay.com/photo/2016/09/07/23/10/cat-1652880_150.jpg",
      webformatURL:
        "https://pixabay.com/get/e833b40d20fc013ed95c4518b74b4f9fe175e5d604b014409df3c77da3ecb2_640.jpg",
      imageWidth: 2116,
      user_id: 686414,
      user: "Alexas_Fotos",
      type: "photo",
      id: 1652880,
      userImageURL:
        "https://cdn.pixabay.com/user/2018/02/07/15-07-53-496_250x250.jpg",
      imageHeight: 1698
    },
    {
      previewHeight: 95,
      likes: 39,
      favorites: 19,
      tags: "cat, red, sleep",
      webformatHeight: 408,
      views: 2509,
      webformatWidth: 640,
      previewWidth: 150,
      comments: 16,
      downloads: 1663,
      pageURL:
        "https://pixabay.com/en/cat-red-sleep-cute-mackerel-tiger-3080966/",
      previewURL:
        "https://cdn.pixabay.com/photo/2018/01/13/23/28/cat-3080966_150.jpg",
      webformatURL:
        "https://pixabay.com/get/ea35b90f21f2073ed95c4518b74b4f9fe175e5d604b014409df3c77da3ecb2_640.jpg",
      imageWidth: 5047,
      user_id: 686414,
      user: "Alexas_Fotos",
      type: "photo",
      id: 3080966,
      userImageURL:
        "https://cdn.pixabay.com/user/2018/02/07/15-07-53-496_250x250.jpg",
      imageHeight: 3223
    },
    {
      previewHeight: 99,
      likes: 242,
      favorites: 217,
      tags: "cat, wink, funny",
      webformatHeight: 425,
      views: 69715,
      webformatWidth: 640,
      previewWidth: 150,
      comments: 39,
      downloads: 36848,
      pageURL: "https://pixabay.com/en/cat-wink-funny-fur-animal-red-1333926/",
      previewURL:
        "https://cdn.pixabay.com/photo/2016/04/16/23/07/cat-1333926_150.jpg",
      webformatURL:
        "https://pixabay.com/get/e836b20c21f6073ed95c4518b74b4f9fe175e5d604b014409df3c77da3ecb2_640.jpg",
      imageWidth: 2808,
      user_id: 686414,
      user: "Alexas_Fotos",
      type: "photo",
      id: 1333926,
      userImageURL:
        "https://cdn.pixabay.com/user/2018/02/07/15-07-53-496_250x250.jpg",
      imageHeight: 1867
    }
  ],
  total: 5
};
