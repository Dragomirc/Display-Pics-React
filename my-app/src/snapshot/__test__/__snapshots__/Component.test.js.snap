// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders 4 x ColorCards 1`] = `
<ul
  className="sc-bwzfXH fRFHuB"
>
  <li
    className="sc-bdVaJa bBrxzW"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#CD5334",
      }
    }
  />
  <li
    className="sc-bdVaJa bBrxzW"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#FFFD66",
      }
    }
  />
  <li
    className="sc-bdVaJa bBrxzW"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#C2F970",
      }
    }
  />
  <li
    className="sc-bdVaJa bBrxzW"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#007EA7",
      }
    }
  />
</ul>
`;

exports[`App renders without crashing 1`] = `
<div
  style={
    Object {
      "maxWidth": 640,
    }
  }
>
  <div
    className="sc-ifAKCX kOzlHD"
  >
    <h1
      className="sc-EHOje cGYXBP"
    >
       Choose a color 
    </h1>
  </div>
  <div>
    <ul
      className="sc-bwzfXH fRFHuB"
    >
      <li
        className="sc-bdVaJa bBrxzW"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#CD5334",
          }
        }
      />
      <li
        className="sc-bdVaJa bBrxzW"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#FFFD66",
          }
        }
      />
      <li
        className="sc-bdVaJa bBrxzW"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#C2F970",
          }
        }
      />
      <li
        className="sc-bdVaJa bBrxzW"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#007EA7",
          }
        }
      />
    </ul>
  </div>
  <div>
    <ul
      className="sc-bxivhb bpAJSv"
    />
  </div>
</div>
`;

exports[`App should render with 5 x ImageCard components when the state is updated 1`] = `
<div
  style={
    Object {
      "maxWidth": 640,
    }
  }
>
  <div
    className="sc-ifAKCX kOzlHD"
  >
    <h1
      className="sc-EHOje cGYXBP"
    >
       Choose a color 
    </h1>
  </div>
  <div>
    <ul
      className="sc-bwzfXH fRFHuB"
    >
      <li
        className="sc-bdVaJa bBrxzW"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#CD5334",
          }
        }
      />
      <li
        className="sc-bdVaJa bBrxzW"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#FFFD66",
          }
        }
      />
      <li
        className="sc-bdVaJa bBrxzW"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#C2F970",
          }
        }
      />
      <li
        className="sc-bdVaJa bBrxzW"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#007EA7",
          }
        }
      />
    </ul>
  </div>
  <div>
    <ul
      className="sc-bxivhb bpAJSv"
    >
      <li
        className="sc-htpNat dcPeku"
      >
        <a
          href="https://pixabay.com/en/domestic-cat-little-cute-kitten-3133620/"
        >
          <img
            alt="Reflecting domestic, cat, little"
            src="https://cdn.pixabay.com/photo/2018/02/06/00/20/domestic-3133620_150.jpg"
          />
        </a>
      </li>
      <li
        className="sc-htpNat dcPeku"
      >
        <a
          href="https://pixabay.com/en/cat-red-cute-mackerel-tiger-sweet-3080965/"
        >
          <img
            alt="Reflecting cat, red, cute"
            src="https://cdn.pixabay.com/photo/2018/01/13/23/28/cat-3080965_150.jpg"
          />
        </a>
      </li>
      <li
        className="sc-htpNat dcPeku"
      >
        <a
          href="https://pixabay.com/en/cat-red-cute-mackerel-tiger-sweet-1652880/"
        >
          <img
            alt="Reflecting cat, red, cute"
            src="https://cdn.pixabay.com/photo/2016/09/07/23/10/cat-1652880_150.jpg"
          />
        </a>
      </li>
      <li
        className="sc-htpNat dcPeku"
      >
        <a
          href="https://pixabay.com/en/cat-red-sleep-cute-mackerel-tiger-3080966/"
        >
          <img
            alt="Reflecting cat, red, sleep"
            src="https://cdn.pixabay.com/photo/2018/01/13/23/28/cat-3080966_150.jpg"
          />
        </a>
      </li>
      <li
        className="sc-htpNat dcPeku"
      >
        <a
          href="https://pixabay.com/en/cat-wink-funny-fur-animal-red-1333926/"
        >
          <img
            alt="Reflecting cat, wink, funny"
            src="https://cdn.pixabay.com/photo/2016/04/16/23/07/cat-1333926_150.jpg"
          />
        </a>
      </li>
    </ul>
  </div>
</div>
`;

exports[`ColorCard renders an li element 1`] = `
<li
  className="sc-bdVaJa bBrxzW"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": undefined,
    }
  }
/>
`;

exports[`ImageCard renders an li with one image child and one anchor element 1`] = `
<li
  className="sc-htpNat dcPeku"
>
  <a
    href={undefined}
  >
    <img
      alt="Reflecting undefined"
      src={undefined}
    />
  </a>
</li>
`;
